ci/typing.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -euo pipefail
+
+poetry run mypy \
```

Comment:
This project uses pyright for types - though I should probably make that clear in the readme and add it as an explicit dependency, as right now it just works when using vscode